modules/gcp/storage.go

```diff
@@ -0,0 +1,94 @@
+package gcp
+
+import (
+	"context"
+	"testing"
+
+	"cloud.google.com/go/storage"
+	"github.com/gruntwork-io/terratest/modules/logger"
+)
+
+// CreateStorageBucket creates a Google Cloud bucket with the given BucketAttrs. Note that Google Storage bucket names must be globally unique.
+func CreateStorageBucket(t *testing.T, projectID string, name string, attr *storage.BucketAttrs) {
+	err := CreateStorageBucketE(t, projectID, name, attr)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+// CreateStorageBucketE creates a Google Cloud bucket with the given BucketAttrs. Note that Google Storage bucket names must be globally unique.
+func CreateStorageBucketE(t *testing.T, projectID string, name string, attr *storage.BucketAttrs) error {
+	logger.Logf(t, "Creating bucket %s", name)
+
+	ctx := context.Background()
+
+	// Creates a client.
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Creates a Bucket instance.
+	bucket := client.Bucket(name)
+
+	// Creates the new bucket.
+	return bucket.Create(ctx, projectID, attr)
+}
+
+// DeleteStorageBucket destroys the Google Storage bucket.
+func DeleteStorageBucket(t *testing.T, name string) {
+	err := DeleteStorageBucketE(t, name)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+// DeleteStorageBucketE destroys the S3 bucket in the given region with the given name.
+func DeleteStorageBucketE(t *testing.T, name string) error {
+	logger.Logf(t, "Deleting bucket %s", name)
+
+	ctx := context.Background()
+
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	return client.Bucket(name).Delete(ctx)
+}
+
+// AssertStorageBucketExists checks if the given storage bucket exists and fails the test if it does not.
+func AssertStorageBucketExists(t *testing.T, name string) {
+	err := AssertStorageBucketExistsE(t, name)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+// AssertStorageBucketExistsE checks if the given storage bucket exists and returns an error if it does not.
+func AssertStorageBucketExistsE(t *testing.T, name string) error {
+	logger.Logf(t, "Finding bucket %s", name)
+
+	ctx := context.Background()
+
+	// Creates a client.
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Creates a Bucket instance.
+	bucket := client.Bucket(name)
+
+	if _, err := bucket.Attrs(ctx); err != nil {
+		// ErrBucketNotExist
+		return err
+	}
+
+	it := bucket.Objects(ctx, nil)
+	if _, err := it.Next(); err == storage.ErrBucketNotExist {
+		return err
+	}
```

Comment:
Hi @brikis98, I spent a while digging through their Go & Java SDK, but it didn't turn up much luck. It appears some calls e.g: `client.Bucket(name)` create an object but don't actually do any activity across the network.

I stumbled on their own integration tests and adapted a bit of the code from there.

You might want to take a look: https://github.com/GoogleCloudPlatform/google-cloud-go/blob/de879f7be552d57556875b8aaa383bce9396cc8c/storage/integration_test.go#L1231

I thought it might be a tad easier and theres still the possibilty I'm on the wrong track.

Reply:
I had the same reaction as @brikis98, though I agree the API here is confusing. Let's add a comment clarifying the uncertainty and linking back to the URL you posted above. If this passes tests, I think it's good enough for now.

Reply:
@josh-padnick ok sure